pkg/objstore/objstore.go

```diff
@@ -172,29 +232,45 @@ func NopCloserWithSize(r io.Reader) io.ReadCloser {
 
 // UploadDir uploads all files in srcdir to the bucket with into a top-level directory
 // named dstdir. It is a caller responsibility to clean partial upload in case of failure.
-func UploadDir(ctx context.Context, logger log.Logger, bkt Bucket, srcdir, dstdir string) error {
+func UploadDir(ctx context.Context, logger log.Logger, bkt Bucket, srcdir, dstdir string, options ...UploadDirOption) error {
 	df, err := os.Stat(srcdir)
+	opts := applyUploadDirOptions(options...)
+	g, ctx := errgroup.WithContext(ctx)
```

Comment:
`The derived Context is canceled the first time a function passed to Go returns a non-nil error or the first time Wait returns, whichever occurs first.`

Reply:
👍